flow/activities/flowable.go

```diff
@@ -801,3 +801,95 @@ func (a *FlowableActivity) AddTablesToPublication(ctx context.Context, cfg *prot
 	}
 	return err
 }
+
+func (a *FlowableActivity) LoadPeer(ctx context.Context, peerName string) (*protos.Peer, error) {
+	row := a.CatalogPool.QueryRow(ctx, `
+		SELECT name, type, options
+		FROM peers
+		WHERE name = $1`, peerName)
+
+	var peer protos.Peer
+	var peerOptions []byte
+	if err := row.Scan(&peer.Name, &peer.Type, &peerOptions); err != nil {
+		return nil, fmt.Errorf("failed to load peer: %w", err)
+	}
+
+	switch peer.Type {
+	case protos.DBType_BIGQUERY:
```

Comment:
this switch looks pretty unfortunate. Could probably collapse it using `reflect`, but then you still need to have mapping from protos.DBType to reflect.Type